components/InventoryItemListItem/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { TableRow, TableCell, Chip } from '@mui/material'
+import { InventoryItem } from 'utils/types'
+import WarningIcon from '@mui/icons-material/Warning'
+
+interface InventoryItemListItemProps {
+  inventoryItem: InventoryItem
+}
+
+export default function InventoryItemListItem({
+  inventoryItem,
+}: InventoryItemListItemProps) {
+  const getCategory = (inventoryItem: InventoryItem) => {
+    // needed to deal with the possible union types defined in the database schema
+    if (
+      typeof inventoryItem.itemDefinition === 'string' ||
+      !inventoryItem.itemDefinition.category
+    ) {
+      return ' '
+    }
+
+    if (typeof inventoryItem.itemDefinition.category === 'string') {
```

Comment:
It'll be a moot point soon, but this `if` is redundant

Reply:
When I remove it, the following return statement (return inventoryItem.itemDefinition.category.name) gives an error because it is not guaranteed that inventoryItem.itemDefinition.category is an object with the property name; it could just be a string. Am I missing something?
Though you are right, whenever the types get fixed, it won't be needed anymore and can be removed.

Reply:
Ah, my mistake.